PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -0,0 +1,687 @@
+    histos.add("Event/h1d_lambda_multiplicity", "#Lambda - Multiplicity", kTH1I, {axisMult});
+    histos.add("Event/h1d_antilambda_multiplicity", "#bar{#Lambda} - Multiplicity", kTH1I, {axisMult});
+
+    // Lambda
+    histos.add("Lambda/h1d_pt", "p_{T}-distribution", kTH1F, {axisPt});
+    histos.add("Lambda/h1d_eta", "#eta-distribution", kTH1F, {axisEta});
+    histos.add("Lambda/h1d_y", "Rapidity", kTH1F, {axisRap});
+    histos.add("Lambda/h1d_phi", "#phi-distribution", kTH1F, {axisPhi});
+    histos.add("Lambda/h1d_inv_mass", "M_{p#pi}", kTH1F, {axisMass});
+
+    // Anti-Lambda
+    histos.addClone("Lambda/", "AntiLambda/");
+
+    // single and two particle densities
+    histos.add("Lambda_Mass/h2d_n1_LaP", "#rho_{1}^{#Lambda}", kTH2D, {axisRap, axisPhi});
+    histos.add("Lambda_Mass/h2d_n1_LaM", "#rho_{1}^{#bar{#Lambda}}", kTH2D, {axisRap, axisPhi});
+    histos.add("Lambda_Mass/h2d_n2_LaP_LaM", "#rho_{2}^{#Lambda - #bar{#Lambda}}", kTH2D, {axisRapPhi, axisRapPhi});
+    histos.add("Lambda_Mass/h2d_n2_LaP_LaP", "#rho_{2}^{#Lambda - #Lambda}", kTH2D, {axisRapPhi, axisRapPhi});
+    histos.add("Lambda_Mass/h2d_n2_LaM_LaM", "#rho_{2}^{#bar{#Lambda} - #bar{#Lambda}}", kTH2D, {axisRapPhi, axisRapPhi});
+
+    histos.addClone("Lambda_Mass/", "Lambda_Right/");
+    histos.addClone("Lambda_Mass/", "Lambda_Left/");
+  }
+
+  template <int mode, typename V>
+  void fillHistos(V const& v)
+  {
+
+    static constexpr std::string_view sub_dir[] = {"Lambda/", "AntiLambda/"};
+
+    histos.fill(HIST(sub_dir[mode]) + HIST("h1d_pt"), v.pt());
+    histos.fill(HIST(sub_dir[mode]) + HIST("h1d_eta"), v.eta());
+    histos.fill(HIST(sub_dir[mode]) + HIST("h1d_y"), v.y());
+    histos.fill(HIST(sub_dir[mode]) + HIST("h1d_phi"), v.phi());
+    histos.fill(HIST(sub_dir[mode]) + HIST("h1d_inv_mass"), v.mass());
+  }
+
+  template <int mode, int hist, typename V>
+  void partSingle(V& p)
+  {
+    static constexpr std::string_view sub_dir_type[] = {"Lambda_Mass/", "Lambda_Right/", "Lambda_Left/"};
+    static constexpr std::string_view sub_dir_hist[] = {"h2d_n1_LaP", "h2d_n1_LaM"};
+
+    histos.fill(HIST(sub_dir_type[mode]) + HIST(sub_dir_hist[hist]), p.y(), p.phi());
+  }
+
+  template <int mode, int hist, typename U, typename V>
+  void partPair(U& p1, V& p2)
+  {
+
+    static constexpr std::string_view sub_dir_type[] = {"Lambda_Mass/", "Lambda_Right/", "Lambda_Left/"};
+    static constexpr std::string_view sub_dir_hist[] = {"h2d_n2_LaP_LaM", "h2d_n2_LaP_LaP", "h2d_n2_LaM_LaM"};
+
+    float nrapbins = static_cast<float>(cfg_nRapBins);
+    float kminrap = static_cast<float>(cfg_Rap_Min);
+    float kmaxrap = static_cast<float>(cfg_Rap_Max);
+    float nphibins = static_cast<float>(cfg_nPhiBins);
+    float kminphi = static_cast<float>(cfg_Phi_Min);
+    float kmaxphi = static_cast<float>(cfg_Phi_Max);
+
+    float rapbinwidth = (kmaxrap - kminrap) / nrapbins;
+    float phibinwidth = (kmaxphi - kminphi) / nphibins;
```

Comment:
This is a time consuming task for doing it for each pair specially because it is not needed
I would suggest doing it once at the `init` time and defining/storing the needed results at the task level